server.js

```diff
@@ -650,34 +650,56 @@ proxyForAuditing.on('error', function(err, req, res) {
   serverUtils.serverError(JSON.stringify(err), req, res);
 });
 
-ddocExtraction.run(function(err) {
+var couchdbCheck = function(callback) {
+  db.request({}, function(err, body) {
+    if (err) {
+      callback(err);
+    }
+
+    var semvers = body.version && body.version.match(/(\d+)\.\d+\.\d+/);
+    if (!semvers || semvers[1] !== '2') {
+      callback('Only CouchDB 2.x is supported, v' + body.version + ' detected.');
+    }
+
+    callback();
+  });
+};
+
+couchdbCheck(function(err) {
```

Comment:
This is too deeply nested now. It was too deeply nested before, but now it's just crazy! Maybe use async.series or something.